### PR TITLE
EKF3 updates for opticalFlow pages

### PR DESCRIPTION
**Pull request by zhouzhiwen2000**

### CODE REVIEW

File: common/source/docs/common-cheerson-cxof.rst [hunk 1]
```diff
@@ -38,11 +38,30 @@ Connection to Autopilot
...
 - Set :ref:`FLOW_TYPE <FLOW_TYPE>` = 4
 - Set :ref:`SERIAL2_PROTOCOL <SERIAL2_PROTOCOL>` = 18 if using Serial2/Telem2, if connected to another serial port use the corresponding SERIALx_PROTOCOL parameter
+
+- For EKF2:
+- Set :ref:`EK3_ENABLE <EK2_ENABLE>` = 1 to enable EKF2 (for ArduPilot 4.1 (and higher))
+- Set :ref:`EK2_ENABLE <EK3_ENABLE>` = 0 to disable EKF3 (for ArduPilot 4.1 (and higher))
 - Set :ref:`EK2_FLOW_DELAY <EK2_FLOW_DELAY>` = 80
 - Optionally set :ref:`EK2_GPS_TYPE <EK2_GPS_TYPE>` = 3 to force the EKF to only use the optical flow sensor and not use the GPS
 
+- For EKF3: (for ArduPilot 4.1 (and higher))
```
Comment (Hwurzburg) [inline common/source/docs/common-cheerson-cxof.rst:51]:
```suggestion
- For EKF3:  EKF3 is enabled and used by default in ArduPilot firmware 4.1 and higher
```

File: common/source/docs/common-cheerson-cxof.rst [hunk 2]
```diff
@@ -38,11 +38,30 @@ Connection to Autopilot
...
+- Set :ref:`EK2_ENABLE <EK2_ENABLE>` = 0 to disable EKF2
```
Comment (Hwurzburg) [inline common/source/docs/common-cheerson-cxof.rst:52]: remove this...not required to disable EKF2

File: common/source/docs/common-openmv-optflow.rst [hunk 3]
```diff
@@ -32,9 +32,27 @@ Connection to Autopilot
 - Set :ref:`FLOW_TYPE <FLOW_TYPE>` = 5 (MAVLink)
 - Set :ref:`SERIAL2_PROTOCOL <SERIAL2_PROTOCOL>` = 1 (MAVLink) if using Serial2/Telem2, if connected to another serial port use the corresponding SERIALx_PROTOCOL parameter
 - Set :ref:`SERIAL2_BAUD <SERIAL2_BAUD>` = 115
+
+- For EKF2:
+- Set :ref:`EK3_ENABLE <EK2_ENABLE>` = 1 to enable EKF2 (for ArduPilot 4.1 (and higher))
```
Comment (Hwurzburg) [inline common/source/docs/common-openmv-optflow.rst:37]:
```suggestion
- Verify :ref:`EK2_ENABLE <EK2_ENABLE>` = 1 is enabling EKF2
```

File: common/source/docs/common-openmv-optflow.rst [hunk 4]
```diff
@@ -32,9 +32,27 @@ Connection to Autopilot
...
+- Set :ref:`EK2_ENABLE <EK3_ENABLE>` = 0 to disable EKF3 (for ArduPilot 4.1 (and higher))
```
Comment (Hwurzburg) [inline common/source/docs/common-openmv-optflow.rst:38]: remove

File: common/source/docs/common-openmv-optflow.rst [hunk 5]
```diff
@@ -32,9 +32,27 @@ Connection to Autopilot
...
 - Set :ref:`EK2_FLOW_DELAY <EK2_FLOW_DELAY>` = 80
 - Optionally set :ref:`EK2_GPS_TYPE <EK2_GPS_TYPE>` = 3 to force the EKF to only use the optical flow sensor and not use the GPS
 
+- For EKF3: (for ArduPilot 4.1 (and higher))
```
Comment (Hwurzburg) [inline common/source/docs/common-openmv-optflow.rst:42]:
```suggestion
- For EKF3: EKF3 is enabled and used by default in ArduPilot firmware 4.1 and higher
```

File: common/source/docs/common-cheerson-cxof.rst [hunk 6]
```diff
@@ -38,11 +38,30 @@ Connection to Autopilot
...
+- Set :ref:`EK3_ENABLE <EK3_ENABLE>` = 1 to enable EKF3
```
Comment (Hwurzburg) [inline common/source/docs/common-cheerson-cxof.rst:53]:
```suggestion
-  Verify  that ref:`EK3_ENABLE <EK3_ENABLE>` = 1 , enabling EKF3
```

File: common/source/docs/common-openmv-optflow.rst [hunk 7]
```diff
@@ -32,9 +32,27 @@ Connection to Autopilot
...
+- Set :ref:`EK2_ENABLE <EK2_ENABLE>` = 0 to disable EKF2
```
Comment (Hwurzburg) [inline common/source/docs/common-openmv-optflow.rst:43]: remove

File: common/source/docs/common-openmv-optflow.rst [hunk 8]
```diff
@@ -32,9 +32,27 @@ Connection to Autopilot
...
+- Set :ref:`EK3_ENABLE <EK3_ENABLE>` = 1 to enable EKF3
```
Comment (Hwurzburg) [inline common/source/docs/common-openmv-optflow.rst:44]:
```suggestion
-  Verify that :ref:`EK3_ENABLE <EK3_ENABLE>` = 1, enabling EKF2
```

File: common/source/docs/common-upixels-upflow.rst [hunk 9]
```diff
@@ -37,9 +37,27 @@ Connection to Autopilot
 - Connect the sensor's VCC to a 3.3V or 5V power source.
 - Set :ref:`FLOW_TYPE <FLOW_TYPE>` = 8
 - Set :ref:`SERIAL2_PROTOCOL <SERIAL2_PROTOCOL>` = 18 if using Serial2/Telem2, if connected to another serial port use the corresponding SERIALx_PROTOCOL parameter
+
+- For EKF2:
+- Set :ref:`EK3_ENABLE <EK2_ENABLE>` = 1 to enable EKF2 (for ArduPilot 4.1 (and higher))
+- Set :ref:`EK2_ENABLE <EK3_ENABLE>` = 0 to disable EKF3 (for ArduPilot 4.1 (and higher))
```
Comment (Hwurzburg) [inline common/source/docs/common-upixels-upflow.rst:43]: remove

File: common/source/docs/common-upixels-upflow.rst [hunk 10]
```diff
@@ -37,9 +37,27 @@ Connection to Autopilot
...
 - Set :ref:`EK2_FLOW_DELAY <EK2_FLOW_DELAY>` = 10
 - Optionally set :ref:`EK2_GPS_TYPE <EK2_GPS_TYPE>` = 3 to force the EKF to only use the optical flow sensor and not use the GPS
 
+- For EKF3: (for ArduPilot 4.1 (and higher))
+- Set :ref:`EK2_ENABLE <EK2_ENABLE>` = 0 to disable EKF2
```
Comment (Hwurzburg) [inline common/source/docs/common-upixels-upflow.rst:48]: remove

File: common/source/docs/common-upixels-upflow.rst [hunk 11]
```diff
@@ -37,9 +37,27 @@ Connection to Autopilot
 - Connect the sensor's VCC to a 3.3V or 5V power source.
 - Set :ref:`FLOW_TYPE <FLOW_TYPE>` = 8
 - Set :ref:`SERIAL2_PROTOCOL <SERIAL2_PROTOCOL>` = 18 if using Serial2/Telem2, if connected to another serial port use the corresponding SERIALx_PROTOCOL parameter
+
+- For EKF2:
+- Set :ref:`EK3_ENABLE <EK2_ENABLE>` = 1 to enable EKF2 (for ArduPilot 4.1 (and higher))
```
Comment (Hwurzburg) [inline common/source/docs/common-upixels-upflow.rst:42]:
```suggestion
- Verify that :ref:`EK2_ENABLE <EK2_ENABLE>` = 1, enabling EKF2
```

File: common/source/docs/common-upixels-upflow.rst [hunk 12]
```diff
@@ -37,9 +37,27 @@ Connection to Autopilot
...
+- Set :ref:`EK2_ENABLE <EK3_ENABLE>` = 0 to disable EKF3 (for ArduPilot 4.1 (and higher))
 - Set :ref:`EK2_FLOW_DELAY <EK2_FLOW_DELAY>` = 10
 - Optionally set :ref:`EK2_GPS_TYPE <EK2_GPS_TYPE>` = 3 to force the EKF to only use the optical flow sensor and not use the GPS
 
+- For EKF3: (for ArduPilot 4.1 (and higher))
```
Comment (Hwurzburg) [inline common/source/docs/common-upixels-upflow.rst:47]:
```suggestion
- For EKF3: EKF3 is enabled and used by default in ArduPilot firmware 4.1 and higher
```

File: common/source/docs/common-upixels-upflow.rst [hunk 13]
```diff
@@ -37,9 +37,27 @@ Connection to Autopilot
...
+- Set :ref:`EK2_ENABLE <EK2_ENABLE>` = 0 to disable EKF2
+- Set :ref:`EK3_ENABLE <EK3_ENABLE>` = 1 to enable EKF3
```
Comment (Hwurzburg) [inline common/source/docs/common-upixels-upflow.rst:49]:
```suggestion
-  Verify that :ref:`EK3_ENABLE <EK3_ENABLE>` = 1, enabling EKF2
```

**Review by Hwurzburg — CHANGES_REQUESTED** [review]
@rmackay9 you will need to verify the setup conditions....thanks

**Review by Hwurzburg — APPROVED** [review]

**Comment by zhouzhiwen2000** [comment]
Is there anything else that needs refining?

**Comment by Hwurzburg** [comment]
just awaiting @rmackay9 approval

**Comment by zhouzhiwen2000** [comment]
> just awaiting @rmackay9 approval

OK, i thought there was something else to modify.😂

**Review by tridge — APPROVED** [review]

**Comment by Hwurzburg** [comment]
approved and merging

**Pull request merged**